shared/actions/login/index.js

```diff
@@ -458,79 +368,99 @@ export function addANewDevice () : AsyncAction {
   }
 }
 
-function makeKex2IncomingMap (dispatch, getState, provisionMethod, userPassTitle, userPassSubtitle) : incomingCallMapType {
+function makeKex2IncomingMap (dispatch, getState/*, provisionMethod, userPassTitle, userPassSubtitle*/) : incomingCallMapType {
   return {
-    'keybase.1.provisionUi.chooseProvisioningMethod': (param, response) => {
-      response.result(provisionMethod)
-    },
     'keybase.1.loginUi.getEmailOrUsername': (param, response) => {
-      const {username} = getState().login.userPass
-      if (!username) {
-        dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
-          const {username} = getState().login.userPass
-          response.result(username)
-        }))
-      } else {
-        response.result(username)
+      const props = {
+        onSubmit: usernameOrEmail => response.result(usernameOrEmail)
       }
+
+      dispatch(routeAppend({
+        parseRoute: {componentAtTop: {component: UsernameOrEmail, props}}
+      }))
+    },
+    'keybase.1.provisionUi.chooseDevice': ({devices}, response) => {
+      const props = {
+        devices,
+        onSelect: deviceID => {
+          const type: deviceRole = devices[devices.findIndex(d => d.deviceID === deviceID)].type
+          const role = {
+            mobile: Constants.codePageDeviceRoleExistingPhone,
+            computer: Constants.codePageDeviceRoleExistingComputer
```

Comment:
Yes it changed

On Friday, March 4, 2016, Marco Munizaga <notifications@github.com> wrote:

> In shared/actions/login/index.js
> <https://github.com/keybase/client/pull/2228#discussion_r55112397>:
>
> >        }
> > +
> > +      dispatch(routeAppend({
> > +        parseRoute: {componentAtTop: {component: UsernameOrEmail, props}}
> > +      }))
> > +    },
> > +    'keybase.1.provisionUi.chooseDevice': ({devices}, response) => {
> > +      const props = {
> > +        devices,
> > +        onSelect: deviceID => {
> > +          const type: deviceRole = devices[devices.findIndex(d => d.deviceID === deviceID)].type
> > +          const role = {
> > +            mobile: Constants.codePageDeviceRoleExistingPhone,
> > +            computer: Constants.codePageDeviceRoleExistingComputer
>
> are we sure the type here is 'mobile' | 'computer' ? I thought I saw
> 'mobile' | 'desktop' somewhere.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/keybase/client/pull/2228/files#r55112397>.
>